client/src/components/NewAttractionForm.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import { Redirect } from "react-router-dom";
+import translateServerErrors from "./../services/translateServerErrors.js";
+import ErrorList from "./layout/ErrorList";
+import FormError from "./layout/FormError";
+
+const NewAttractionForm = (props) => {
+  const [newAttraction, setNewAttraction] = useState({
+    name: "",
+    description: "",
+  });
+  const [errors, setErrors] = useState([])
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const addNewAttraction = async () => {
+    try {
+      const response = await fetch("/api/v1/attractions", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(newAttraction),
+      });
+      const body = await response.json();
+      if (!response.ok) {
+        if (response.status === 422) {
+          const newErrors = translateServerErrors(body.errors);
+          return setErrors(newErrors);
+        }
+        throw new Error(`${response.status} (${response.statusText})`);
+      } else {
+        setShouldRedirect(true);
+      }
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+  const handleInputChange = (event) => {
```

Comment:
I would add an empty line here to separate the functions, mainly for readability purposes